perceval/components/abstract_processor.py

```diff
@@ -284,6 +297,8 @@ def _compose_processor(self, connector: ModeConnector, processor, keep_port: boo
         # Compute new herald positions
         n_new_heralds = connector.add_heralded_modes(mode_mapping)
         self._n_heralds += n_new_heralds
+        self._mode_type += [ModeType.HERALD] * n_new_heralds
+        self._detectors += [None] * n_new_heralds
```

Comment:
So we simply erase the detectors that are on the heralds of the composed processor?